src/shared/components/plots/MultipleCategoryBarPlotUtils.ts

```diff
@@ -98,6 +98,72 @@ export function sortDataByCategory<D>(
         }
     });
 }
+function sortDataByOption(
+    data: IMultipleCategoryBarPlotData[],
+    sortByOption: string
+): IMultipleCategoryBarPlotData[] {
+    if (sortByOption == 'SortByTotalSum') {
+        const majorCategoryCounts: any = {};
+
+        data.forEach(item => {
+            item.counts.forEach(countItem => {
+                const { majorCategory, count } = countItem;
+                if (!majorCategoryCounts[majorCategory]) {
+                    majorCategoryCounts[majorCategory] = 0;
+                }
+                majorCategoryCounts[majorCategory] += count;
+            });
+        });
+        const sortedMajorCategories = Object.keys(majorCategoryCounts).sort(
+            (a, b) => majorCategoryCounts[b] - majorCategoryCounts[a]
+        );
+
+        const reorderCounts = (counts: any) => {
```

Comment:
instead of making this function inside, lets extract this into a new function outside for clarity and reuse